txlib/api/base.py

```diff
@@ -1,45 +1,126 @@
+    """Base class for Transifex models.
 
     Each model has a list of fields. The model works as a proxy between the
-    local application and the remote Tx server.
+    local application and the remote Transifex server.
 
-    The user of the class can either create a local-only instance using the
-    staticmethod ``get`` or retrieve a remote object whenever he accesses
-    an attribute that has no value.
+    The user of the class can:
+      a) retrieve an existing remote instance of the model
+         by using the static method `MyModel.get(...)`
+      b) create a new local instance with a set of populated fields
+         and call `save()` in order to save it to the remote server
+      c) delete an existing remote instance, by first creating a local instance
+         and all necessary attributes that identify the object, and then
+         calling `delete()` on it.
     """
 
+    # The URI prefix of all API endpoints for this model
     _prefix = ''
-    _path_to_collection = ''    # shouldn't start with a slash
-    _path_to_item = ''          # shouldn't start with a slash
 
-    read_only_fields = set()
-    write_also_fields = set()
-    mandatory_fields = set()
-    fields = read_only_fields | write_also_fields
-    url_fields = {}
+    # The URI for retrieving a collection of multiple items
+    # (shouldn't start with a slash)
+    _path_to_collection = ''
+
+    # The URl for retrieving a single item
+    # (shouldn't start with a slash)
+    _path_to_item = ''
+
+    # All fields defined here will be used for constructing
+    # the URL of the request
+    url_fields = set()
+
+    # These fields can be modified in POST/PUT requests
+    writable_fields = set()
+
+    # Initially False, set to True when an instance of the class is created
+    _is_initialized = False
 
     @classmethod
     def get(cls, **kwargs):
-        """Method to retrieve an object from a Tx server."""
-        url_fields = {}
+        """Retrieve an object by making a GET request to Transifex.
+
+        Each value in `kwargs` that corresponds to a field
+        defined in `self.url_fields` will be used in the URL pth
+        of the request, so that a particular entry of this model
+        is identified and retrieved.
+
+        Raises:
+            txlib.http.exceptions.NotFoundError: if the object with these
+                attributes is not found on the remote server
+            txlib.http.exceptions.ServerError subclass: depending on
+                the particular server response
+
+        Example:
+        # Note: also catch exceptions
+        >>> obj = MyModel.get(attr1=value1, attru2=value2)
+        """
+        fields = {}
         for field in cls.url_fields:
-            url_fields[field] = kwargs.pop(field, None)
-        model = cls(**url_fields)
+            fields[field] = kwargs.pop(field, None)
+
+        # Create an instance of the model class and make the GET request
+        model = cls(**fields)
```

Comment:
Why create a new instance here? Don't we already have one stored in `cls`?

Reply:
This is a static method, so `cls` is the class itself, not the instance of the class.